### PR TITLE
Fix not finding locally installed eslint

### PR DESCRIPTION
**Pull request by piratefsh**
Fixed a couple of issues I encountered while running ESLint-Formatter on MacOS, ST3 Build 3114.

1. ESLint-Formatter was unable to find my locally installed eslint (in node_modules/.bin). After some debugging, it was because this line:

```
normdn = PluginUtils.normalize_path(os.getcwd())
```

was returning the cwd as the package, and recursing upwards from there t look for node_modules (which is non-existent)

```
/Users/myuser/Library/Application Support/Sublime Text 3/Packages
```

I patched it to look for `node_module` relative to the current project's path (see the `project_path` method).

2. The config file did not support relative path to the project folder (I had to specify the full path to my .eslintrc file). I patched it to look for the file relative to project path if the file path specified is not found. (line 82-88).

This is my first time working on a Sublime text plugin, so comments would be most welcome.

### CODE REVIEW

File: ESLint-Formatter.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def run(self, edit):
 
     output = self.run_script_on_file(self.view.file_name())
 
+    # log output for easier debugging
```
Comment (TheSavior) [inline ESLint-Formatter.py]: I'm not sure that we want to commit this. It seems like it would print lots of messages to the sublime console.

**Comment by TheSavior** [comment]
This looks great. Thanks for the commit. Can you add a change to the Readme explaining how it looks for the eslint executable? Now that we have both findup and logic that looks at the first folder in your project configuration, we should probably help make that clear.

File: ESLint-Formatter.py (hunk 1 above)
Comment (piratefsh) [inline ESLint-Formatter.py]: I can look into only printing it when debug mode for SublimeLinter is on. IMO it'll be good to have as the plugin was failing on me silently (with debug mode on) which made it harder to debug the issue.

**Comment by piratefsh** [comment]
@TheSavior thanks. sure, I'll write it up in the README.

File: ESLint-Formatter.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import glob
 import os, sys, subprocess, codecs, webbrowser
 from subprocess import Popen, PIPE
+from SublimeLinter.lint import persist
```
Comment (TheSavior) [inline ESLint-Formatter.py]: SublimeLinter is unrelated to this project and probably doesn't make sense to be a dependency. I previously had assumed that people who used these formatters also had the associated linters installed. That assumption was surprisingly proven wrong.

Comment (piratefsh) [inline ESLint-Formatter.py]: Ah, my bad. That makes sense. For some reason I built up a mental model with SublimeLinter as a dep for the formatter. I removed it and added a debug flag to the settings. Hope that works.

**Comment by TheSavior** [comment]
Thanks for the PR. I'll release a new version in a few hours (I'm not on the correct laptop)

**Comment by TheSavior** [comment]
Released as 2.2.0

**Comment by piratefsh** [comment]
Awesome, thank you for your work!

**Comment by TheSavior** [comment]
Haha, you did all the work! Thanks for contributing.

**Pull request merged**